redash/monitor.py

```diff
@@ -70,3 +74,66 @@ def get_status():
     status['database_metrics']['metrics'] = get_db_sizes()
 
     return status
+
+
+def parse_control_tasks_list(tasks, tasks_state):
+    query_tasks = []
+
+    if tasks is None:
+        return query_tasks 
+
+    for task in itertools.chain(*tasks.values()):
+        data_source_id = None
+        query_id = None
+        user = None
+
+        # TODO: we should support parsing all the tasks, not only query ones.
+        if task['name'] == 'redash.tasks.execute_query':
+            args = ast.literal_eval(task['args'])
```

Comment:
🤢

Reply:
It looks like they changed something in how this works and it no longer works. The good news are that because of this I found a better way to access the data: Celery gives you the option to set your own representation of the arguments, so I piggy back on this to provide a JSON object.